src/citrine/_serialization/properties.py

```diff
@@ -612,28 +686,37 @@ def _serialize(self, value: typing.Any) -> str:
         return self.klass.get_value(value)
 
 
+@lru_cache(maxsize=1024 * 1024)
+def _fields_map(klass: typing.Type) -> typing.Dict[str, Property]:
+    """Compute the properties-relevant fields in a given class."""
+    return {
+        k: v
+        for x in reversed(klass.__mro__)  # Classes at the front trump
+        for k, v in x.__dict__.items() if isinstance(v, Property)
+    }
```

Comment:
Uses a reverse MRO instead of piecemeal pulling things in from two most immediate layers.  Allows for proper method inheritance instead of just a shallow inheritance.